support/on-prem-bootstrap.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+# The purpose of this script is to download the latest stable version of every
+# package in the core-plans repo, tar them up, and upload to S3. It also supports
+# downloading the archive from S3, extracting it, and uploading to a new depot.
+
+set -e
+
+help() {
+    echo "Usage: on-prem-bootstrap.sh {create-archive|populate-depot <DEPOT_URL>}"
+}
+
+exists() {
+  if command -v $1 >/dev/null 2>&1
+  then
+    return 0
+  else
+    return 1
+  fi
+}
+
+required_tools=( aws git curl jq )
+required_vars=( AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY HAB_AUTH_TOKEN )
+upstream_depot="https://bldr.habitat.sh"
+bucket="habitat-on-prem-builder-bootstrap"
+bootstrap_file="on-prem-bootstrap.tar.gz"
+
+for tool in "${required_tools[@]}"
+do
+  if ! exists "$tool"; then
+    echo "Please install $tool and run this script again."
+    exit 1
+  fi
+done
+
+for var in "${required_vars[@]}"
+do
+  if [ -z "${!var}" ]; then
+    echo "Please ensure that $var is exported in your environment and run this script again."
+    exit 1
+  fi
+done
+
+case "$1" in
+  create-archive)
+    core="/tmp/core-plans"
+    tar_file="/tmp/$bootstrap_file"
+    name=$(cat /dev/urandom | tr -cd 'a-f0-9' | head -c 32)
+    tmp_dir="/tmp/$name"
+
+    mkdir -p "$tmp_dir"
+    rm -fr "$core"
+    git clone https://github.com/habitat-sh/core-plans.git "$core"
+    pushd "$core"
+    for p in $(find . -maxdepth 1 -type d -exec basename '{}' \; | sort)
+    do
+      latest=$(curl -H "Accept: application/json" "$upstream_depot/v1/depot/channels/core/stable/pkgs/$p/latest")
+      raw_ident=$(echo "$latest" | jq ".ident")
+
+      if [ "$raw_ident" = "null" ]; then
+        continue
+      fi
+
+      slash_ident=$(echo "$raw_ident" | jq "\"\(.origin)/\(.name)/\(.version)/\(.release)\"" | tr -d '"')
+      dash_ident=$(echo "$raw_ident" | jq "\"\(.origin)-\(.name)-\(.version)-\(.release)\"" | tr -d '"')
+      target=$(echo "$latest" | jq ".target" | tr -d '"')
+
+      curl -H "Accept: application/json" -o $tmp_dir/$dash_ident-$target.hart "$upstream_depot/v1/depot/pkgs/$slash_ident/download"
+    done
+    popd
+
+    tar zcvf $tar_file -C $tmp_dir .
+    aws s3 cp $tar_file s3://$bucket/
+    rm -fr "$tmp_dir"
+    rm "$tar_file"
+    echo "Done."
+    ;;
+  populate-depot)
+    if [ -z "$2" ]; then
+      help
+      exit 1
+    fi
+
+    cache_dir="/hab/cache/artifacts"
```

Comment:
Is this correct?  I could be wrong, but I _believe_ when uploading a hart, it looks for the deps in the same directory as the hart.

If we need to take over the user's artifact cache, we should probably warn and prompt them..

Reply:
It didn't seem to work in my tests, but it's hard to do a "real" test of this because doing so would create a gigantic tarball.  Maybe I'll try it out on an EC2 instance.